src/lightning/data/cache/writer.py

```diff
@@ -0,0 +1,273 @@
+# Copyright The Lightning AI team.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+from typing import Any, Dict, Optional
+
+import numpy as np
+
+from lightning.data.cache.compression import _COMPRESSORS
+from lightning.data.cache.pytree import tree_flatten, treespec_dumps
+from lightning.data.cache.serializers import _SERIALIZERS, Serializer
+from lightning.data.cache.worker import get_worker_info
+from lightning.data.datasets.env import _DistributedEnv, _WorkerEnv
+
+
+def cloud_path(cache_dir: str) -> Optional[str]:
```

Comment:
If you could go over all APIs again and check whether they should be public. I'm not sure we want to expose all these.

Reply:
Will do.